develop/blockchains/custom-blockchains/benchmarking.md

```diff
@@ -1,4 +1,223 @@
+
+## Preparing Your Environment
+
+Before writing benchmarks, you need to ensure the `frame-benchmarking` crate is included in your pallet's `Cargo.toml`:
+
+```toml
+frame-benchmarking = { version = "37.0.0", default-features = false }
+```
+
+You must also ensure that you add the `runtime-benchmarks` feature flag as follows under the `[features]` section of your pallet's `Cargo.toml`:
+
+```toml
+runtime-benchmarks = [
+    "frame-benchmarking/runtime-benchmarks",
+    "frame-support/runtime-benchmarks",
+    "frame-system/runtime-benchmarks",
+    "sp-runtime/runtime-benchmarks",
+]
+```
+
+Lastly, ensure that `frame-benchmarking` is included in `std = []`: 
+
+```toml
+std = [
+    # ...
+    "frame-benchmarking?/std",
+    # ...
+]
+```
+
+Once this is complete, you have the required dependencies for writing benchmarks for your pallet.
+
+## Writing Benchmarks
+
+Create a `benchmarking.rs` file in your pallet's `src/`:
+
+```
+my-pallet/
+├── src/
+│   ├── lib.rs          # Main pallet implementation
+│   └── benchmarking.rs # Benchmarking
+└── Cargo.toml
+```
+
+After which, you may copy the barebones template below to get started:
+
+!!!note "This example is from the pallet template"
+    Take a look at the repository in the [`polkadot-sdk-parachain-template`](https://github.com/paritytech/polkadot-sdk-parachain-template/tree/master/pallets){target=\_blank}
+
+```rust
+--8<-- 'https://raw.githubusercontent.com/paritytech/polkadot-sdk-parachain-template/refs/tags/v0.0.2/pallets/template/src/benchmarking.rs'
+```
+
+The function `do_something` is a placeholder; you must write your own function that tested your extrinsic. Similar to writing unit tests, you have access to the mock runtime, and can use functions such as `whitelisted_caller()` (an account whitelisted for DB reads/writes) to sign transactions in the benchmarking context.
+
+There are a couple practices to notice: 
+
+- The `#[extrinsic_call]` macro is used when calling the extrinsic itself. This macro is a required part of a benchmarking function, [see the Rust docs](https://paritytech.github.io/polkadot-sdk/master/frame_benchmarking/v2/index.html#extrinsic_call-and-block){target=_blank} for more clarification
+- The `assert_eq` expression ensures that the extrinsic is working properly within the benchmark context
+
+## Adding Benchmarks to Runtime
+
+The last step before running the benchmarking tool is to ensure that the benchmarks are configured as part of your runtime.
+
+Create another file in the node's directory called `benchmarks.rs`. This is where the pallets you wish to benchmark will be registered. This file should contain the following macro, which registers all pallets for benchmarking, as well as their respective configurations: 
+
+```rust
+frame_benchmarking::define_benchmarks!(
```

Comment:
https://paritytech.github.io/polkadot-sdk/master/frame_benchmarking/macro.define_benchmarks.html

Is this good enough? can you contribute back anything to this?

Reply:
Maybe an example + slightly clearer description of what that looks like syntactically to better illustrate what is going on, I can open a PR and do this if wanted